pkg/version/version.go

```diff
@@ -191,3 +197,81 @@ func IsReleaseVersion() bool {
 		GetMainVersion() != "" &&
 		!strings.Contains(GetMainVersion(), "-")
 }
+
+// IsPriorToScannerV4 returns true if version represents a version of ACS from prior to the
+// introduction of Scanner V4. Will return an error if cannot determine result.
+func IsPriorToScannerV4(version string) (bool, error) {
+	var err error
+	parsed, err := parseVersion(version)
+	if err != nil {
+		return false, err
+	}
+
+	x := parsed.MarketingMajor
+	y := parsed.EngRelease
+	z := parsed.PatchLevel
+
+	if x < 4 || y < 3 || (y == 3 && (z == "" || z != "x")) {
+		return true, nil
+	}
+	return false, nil
+}
+
+// Variants breaks a version into a series of version strings starting with
+// the most specific to the least specific, stopping at X.Y. It uses
+// parseVersion to validate the format.
+func Variants(version string) ([]string, error) {
+	_, err := parseVersion(version)
+	if err != nil {
+		return nil, err
+	}
+
+	res := []string{version}
+
+	i := strings.LastIndex(version, "-")
+	for i != -1 {
+		res = append(res, version[:i])
+		version = version[:i]
+		i = strings.LastIndex(version, "-")
+	}
+
+	i = strings.LastIndex(version, ".")
+	for i != -1 && strings.Count(version, ".") > 1 {
+		res = append(res, version[:i])
+		version = version[:i]
+		i = strings.LastIndex(version, ".")
+	}
```

Comment:
```suggestion
	for i := strings.LastIndex(version, "."); i != -1 && strings.Count(version, ".") > 1; i = strings.LastIndex(version, ".") {
		res = append(res, version[:i])
		version = version[:i]
	}
```

Reply:
Can't say I am a fan of the longer lines or (seemingly) additional complexity of this change, but will add it.

Reply:
https://github.com/stackrox/stackrox/pull/9803/commits/30c2528cf548279c94bc5378ba3d06de70e6edf8